cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -0,0 +1,120 @@
+package relayer
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	tmCommon "github.com/tendermint/tendermint/libs/common"
+	tmLog "github.com/tendermint/tendermint/libs/log"
+	tmClient "github.com/tendermint/tendermint/rpc/client"
+	tmTypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/peggy/cmd/ebrelayer/events"
+	"github.com/cosmos/peggy/cmd/ebrelayer/txs"
+)
+
+// InitCosmosRelayer : initializes a relayer which witnesses events on the Cosmos network and relays them to Ethereum
+func InitCosmosRelayer(
+	tendermintProvider string,
+	web3Provider string,
+	contractAddress common.Address,
+	rawPrivateKey string,
+) error {
+	logger := tmLog.NewTMLogger(tmLog.NewSyncWriter(os.Stdout))
+	client := tmClient.NewHTTP(tendermintProvider, "/websocket")
+
+	client.SetLogger(logger)
+
+	err := client.Start()
+	if err != nil {
+		logger.Error("Failed to start a client", "err", err)
+		os.Exit(1)
+	}
+
+	defer client.Stop()
+
+	// Subscribe to all tendermint transactions
+	query := "tm.event = 'Tx'"
+
+	out, err := client.Subscribe(context.Background(), "test", query, 1000)
+	if err != nil {
+		logger.Error("Failed to subscribe to query", "err", err, "query", query)
+		os.Exit(1)
+	}
+
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, os.Interrupt, syscall.SIGTERM)
+
+	for {
+		select {
+		case result := <-out:
+			tx, ok := result.Data.(tmTypes.EventDataTx)
+			if !ok {
+				logger.Error("Type casting failed while extracting event data from new tx")
+			}
+
+			logger.Info("New transaction witnessed")
+
+			// Iterate over each event inside of the transaction
+			for _, event := range tx.Result.Events {
+				// Get type of OracleClaim based on the event's type
+				claimType := getOracleClaimType(event.GetType())
+
+				switch claimType {
+				case events.MsgBurn, events.MsgLock:
+					// Parse event data, then package it as a ProphecyClaim and relay to the Ethereum Network
+					err := handleBurnLockMsg(event.GetAttributes(), claimType, web3Provider, contractAddress)
+					if err != nil {
+						return err
+					}
+				case events.Unsupported:
```

Comment:
Deleted, but keeping the switch statement as it will be useful later for different types of Cosmos msgs https://github.com/cosmos/peggy/commit/52fdedf6f0b4e4b1c7cf822ab3519dabba3afa17